packages/dds/map/src/directory.ts

```diff
@@ -1519,6 +1519,16 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
                     this.serializer,
                     posix.join(this.absolutePath, subdirName)),
             );
+
+            const event: IDirectoryValueChanged = {
```

Comment:
For create & delete events, I'd suggest to define and use separate interface

Reply:
And separate event names, of course :)

Reply:
That said, with current structure (of having valueChanged & containedValueChanged events) it would be hard as all events need to be duplicated.

Reply:
You're right - I see the problem with using the same event when keys and subdirectories are handled differently by everything else.